File: frontend/routes/member.route.js
const router = require("express").Router();

const {
  middlewareCheckGoogleProfile,
  middlewareCheckEmail,
} = require('../controllers/oauth-google.controller');

const {
  getMemberRoom,
  postCreateRoom,
  putJoinRoom,
} = require('../controllers/member.controller');

router.get(/\/room\/page(\d+)/, middlewareCheckGoogleProfile, middlewareCheckEmail, getMemberRoom);
router.post("/room/create", middlewareCheckGoogleProfile, middlewareCheckEmail, postCreateRoom);
router.put("/room/join", middlewareCheckGoogleProfile, middlewareCheckEmail, putJoinRoom);

module.exports = router;

File: backend/routes/member.route.go
package routes

import (
	"github.com/gofiber/fiber/v2"
	controller "github.com/simopunkc/chirpbird-v2/controllers"
)

func MemberRoute(app fiber.Router) {
	app.Get("/room/page:pages", controller.GetMemberRoom)
	app.Post("/room/create", controller.PostCreateRoom)
	app.Put("/room/join", controller.PutJoinRoom)
}
